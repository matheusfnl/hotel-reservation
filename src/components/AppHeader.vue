<template>
  <div
    class="row items-center q-mx-auto q-pa-xs"
    :style="{ justifyContent: $q.screen.lt.sm ? 'center' : 'start' }"
    style="max-width: 1600px"
  >
    <h1 class="h1-reset">
      <router-link :to="{ name: 'home' }" class="cursor-pointer">
        <img
          :src="Logo"
          :alt="$t('shared.logo_alt')"
          :style="{ width: $q.screen.lt.sm ? '150px' : '200px' }"
        />
      </router-link>
    </h1>
  </div>
</template>

<script lang="ts" setup>
import Logo from '@/assets/main-logo.svg'
</script>

<style scoped>
.h1-reset {
  all: unset;
  display: block;
}
</style>
