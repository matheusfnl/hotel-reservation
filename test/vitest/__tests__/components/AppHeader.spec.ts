import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'

import AppHeader from '@/components/AppHeader.vue'

const mountWrapper = () => {
  return mount(AppHeader)
}

describe('AppHeader', () => {
  it('should render header', () => {
    const wrapper = mountWrapper()

    expect(wrapper.html()).toMatchSnapshot()
  })
})
