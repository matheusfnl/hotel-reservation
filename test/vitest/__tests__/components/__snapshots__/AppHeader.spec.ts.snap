// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AppHeader > should render header 1`] = `
"<div data-v-48a13218="" class="row items-center q-mx-auto q-pa-xs" style="justify-content: start; max-width: 1600px;">
  <h1 data-v-48a13218="" class="h1-reset">
    <router-link data-v-48a13218="" to="[object Object]" class="cursor-pointer"><img data-v-48a13218="" src="/src/assets/main-logo.svg" alt="Onfly Logo" style="width: 200px;"></router-link>
  </h1>
</div>"
`;
